src/olympia/files/utils.py

```diff
@@ -829,6 +843,48 @@ def extract_zip(source, remove=False, force_fsync=False):
     return tempdir
 
 
+def extract_extension_to_dest(source, dest=None, force_fsync=False):
+    """Extract `source` to `dest`.
+
+    `source` can be an extension or extension source, can be a zip, tar
+    (gzip, bzip) or a search provider (.xml file).
+
+    Note that this doesn't verify the contents of `source` except for
+    that it requires something valid to be extracted.
+
+    :returns: Extraction target directory, if `dest` is `None` it'll be a
+              temporary directory.
+    """
+    target, tempdir = None, None
+
+    if dest is None:
+        target = tempdir = tempfile.mkdtemp(dir=settings.TMP_PATH)
```

Comment:
The temporary-directory fallback isn't used for git-extraction but can be re-used by all the other extract_ methods that we have and potentially be unified in one interface in the future. At least that was what I had in mind :see_no_evil: